src/stubs/sm_entry.s

```diff
@@ -14,9 +14,19 @@ __sm_entry:
     ; not actually be called by an IRQ in which case we might overwrite a stored
     ; stack pointer.
     mov r1, &__sm_tmp
-    # Switch stack.
-    ; __sm_sp is our stackpointer
-    mov #__sm_sp, &__sm_sp_addr
+
+    ; initialize SSA on first entry (if ssa_thread_id is 0x0)
+    ; Use r1 for this check as we just backed it up
+    mov #__sm_ssa_base_addr, r1
+    cmp #0x0, r1
+    jne 1f
+    ; If SSA address was not initialized, do it now
+    ; Technically, this could be set to any SSA but we have just one for now
+    mov #__sm_ssa_base, &__sm_ssa_base_addr
+    
+1:  
+    ; Now switch the stack to the ssa base
+    ; __sm_sp is our stackpointer, it lies at #ssa_base-2
     mov &__sm_sp, r1
     ; initialize sp on first entry
     cmp #0x0, r1
```

Comment:
below it says "safe to handle IRQs now" which is *not* true afais! With the new SSA frame convention, the reti logic itself is _not_ interruptible so the SSA has to be copied in the reg file in case of an IRQ return _before_ IRQs are re-enabled...

Reply:
I moved the safe to handle IRQs to later in the sm_entry and into the reti entry. There we still need to resolve the pop r2 issue but besides that this is resolved I think